### ISSUE TITLE
🐛 BUG: Named slots result in "undefined" as a string in generated elements attributes

### ISSUE THREAD

**Issue by lloydjatkinson** [snowpackjs/astro#1564]
### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

When using **named** slots the output HTML has an erroneous `undefined` in the attributes.

```html
// Example.astro
<slot name="on" />
```

```html
// Usage:
<p slot="on">I'm switched on ✅</p>
```

Results in:

```html
<p undefined="">I'm switched on ✅</p>
```

### Steps to Reproduce

Run the reproduction in the link. I also tried it with a component with a ternary conditionally rendering differents slots and also a simple component that was only a single slot. The `undefined` is generated in both cases.

(Right click, Inspect)

### Link to Minimal Reproducible Example (Optional)

https://stackblitz.com/edit/astro-ajypi4?file=src/components/Example.astro

**Comment by stefanprobst** [comment]
looks like a duplicate of #1453?

**Comment by lloydjatkinson** [comment]
Here's a quick unit test I wrote to demonstrate the problem:

```js
// packages/astro/test/astro-slots.test.js
Slots('Basic named slots create correct attributes', async({ runtime }) => {
  const result = await runtime.load('/');
  assert.ok(!result.error, `build error: ${result.error}`);

  const $ = doc(result.contents);

  if ($('#a')[0].children[0].attribs.undefined === '') {
    throw new Error('slot should not contain redundant attribute')
  }
});
```

```
   FAIL  Slot test  "Basic named slots create correct attributes"
    slot should not contain redundant attribute
```

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
